apps/prairielearn/src/components/SyncProblemButton.tsx

```diff
@@ -1,30 +1,39 @@
-import { escapeHtml, html, unsafeHtml } from '@prairielearn/html';
-
 import { ansiToHtml } from '../lib/chalk.js';
+import { renderHtml } from '../lib/preact-html.js';
 
 export function SyncProblemButton({ output, type }: { output: string; type: 'error' | 'warning' }) {
   const title = type === 'error' ? 'Sync Errors' : 'Sync Warnings';
   const classes =
     type === 'error' ? 'fa-times text-danger' : 'fa-exclamation-triangle text-warning';
 
-  const popoverContent = html`<pre
-    class="text-white rounded p-3 mb-0"
-    style="background-color: black;"
-  >
-${unsafeHtml(ansiToHtml(output))}</pre
-  >`;
+  const popoverContent = renderHtml(
+    <pre class="text-white rounded p-3 mb-0" style="background-color: black;">
+      {ansiToHtml(output, { escapeHtml: false })}
```

Comment:
👀 does this not need `dangerouslySetInnerHtml`? I'm a bit surprised that this works.

Reply:
Ah, it does not in fact work!

<img width="856" height="281" alt="Image" src="https://github.com/user-attachments/assets/2ba3411e-5682-4d64-a6ea-d40164de9ac8" />

I suspect that once you fix this, the `escapeHtml: false` flag becomes unnecessary.

Reply:
<img width="962" height="488" alt="CleanShot 2025-07-22 at 11 33 51@2x" src="https://github.com/user-attachments/assets/06906385-15a4-44e1-927b-dd236d65461c" />